src/test/ui/consts/issue-70114.rs

```diff
@@ -0,0 +1,12 @@
+// build-pass
+
+// Test a ZST enum whose dicriminant is ~0i128. This caused an ICE when casting to a i32.
+
+enum Nums {
+    NegOne = -1,
+}
+
+fn main() {
+    let kind = Nums::NegOne;
+    let _v = kind as i32;
```

Comment:
Could you `assert!` that the result of the conversion is correct?
Also, I think the reason this triggers here is const-prop. So it's probably best to just move this to a `const` explicitly so we can see it runs in const context.

Also the issue says shorter types are a problem, so maybe add a cast to `i8` as well?

Reply:
I think this test should include every single type and they should all end up as `-1`.
Including `i128` which is currently broken even for runtime code (see my other comments).

Reply:
So the test should probably cover const-time and run-time code then.
We can avoid const-prop by feeding values through a function like
```rust
#[inline(never)]
fn black_box<T>(x: T) -> T { x }
```

Reply:
A problem there is you don't need the value of an univariant `enum` to compute the case, because it's ZST.

Reply:
Also don't we just call that function `identity` in tests?

Reply:
We call it different things I think...^^